contrib/native/client/src/clientlib/drillClientImpl.cpp

```diff
@@ -595,6 +611,12 @@ connectionStatus_t DrillClientImpl::validateHandshake(DrillUserProperties* prope
 
     switch(this->m_handshakeStatus) {
         case exec::user::SUCCESS:
+            // Check if client needs auth/encryption and server is not requiring it
```

Comment:
Not too clear about the SASL flow, but I assume that if the server returning SUCCESS is sufficient to assume that there is no auth required by the server.

Reply:
Yes. The control flow goes through the AUTH_REQUIRED case when the server requires auth.